annotationProcessor/src/main/java/org/corfudb/annotations/ObjectAnnotationProcessor.java

```diff
@@ -311,10 +311,24 @@ public void generateProxy(TypeElement classElement)
                             ||
                             x.getModifiers().contains(Modifier.PROTECTED))
                     .forEach(x -> {
-                        if (methodSet.stream().noneMatch(y ->
-                                // If this method is present in the parent, the string
-                                // will match.
-                                y.method.toString().equals(x.toString()))) {
+                        if (methodSet.stream().noneMatch(y -> {
+                            // If this method is present in the parent, we need to check
+                            // the parameters
+                            if (y.method.getSimpleName().equals(x.getSimpleName())) {
+                                if (y.method.getParameters().size() == x.getParameters().size()) {
+                                    // If there are generics, assume a match for now
+                                    // TODO: Properly handle generics
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)